libyul/AssemblyStack.cpp

```diff
@@ -226,6 +226,48 @@ MachineAssemblyObject AssemblyStack::assemble(Machine _machine) const
 	return MachineAssemblyObject();
 }
 
+std::pair<MachineAssemblyObject, MachineAssemblyObject> AssemblyStack::assemble(std::string const& _runtimeName) const
+{
+	yulAssert(m_analysisSuccessful, "");
+	yulAssert(m_parserResult, "");
+	yulAssert(m_parserResult->code, "");
+	yulAssert(m_parserResult->analysisInfo, "");
+
+	evmasm::Assembly assembly;
+	EthAssemblyAdapter adapter(assembly);
+	compileEVM(adapter, false, m_optimiserSettings.optimizeStackAllocation);
+
+	MachineAssemblyObject creationObject;
+	creationObject.bytecode = make_shared<evmasm::LinkerObject>(assembly.assemble());
+	yulAssert(creationObject.bytecode->immutableReferences.empty(), "Leftover immutables.");
+	creationObject.assembly = assembly.assemblyString();
+	creationObject.sourceMappings = make_unique<string>(
+		evmasm::AssemblyItem::computeSourceMapping(
+			assembly.items(),
+			{{scanner().charStream() ? scanner().charStream()->name() : "", 0}}
+		)
+	);
+
+	MachineAssemblyObject runtimeObject;
+
+	// Pick matching runtime assembly
```

Comment:
What if it is not found?

Reply:
same as before this PR, it's ignoring it (the heuristic only did something if there was exactly one sub assembly)

Reply:
But it's different now, the other function was called "guessRuntime"